extensions/amp-inline-gallery/1.0/inline-gallery.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as Preact from '../../../src/preact';
+import {cloneElement, toChildArray, useState} from '../../../src/preact';
+
+/**
+ * @param {!InlineGalleryDef.Props} props
+ * @return {PreactDef.Renderable}
+ */
+export function InlineGallery({children, ...rest}) {
+  const childrenArray = toChildArray(children);
+  const carousel = childrenArray.filter((child) => !!child.props.children);
```

Comment:
This is a very big assumption, imho.

Reply:
Yeah, this is by no means finalized. What's a better way to identify the target carousel? Passing it as a prop doesn't feel intuitive either seeing as it still also has to be a child.
```js
const myCarousel = <BaseCarousel ...></BaseCarousel>
return (
  <InlineGallery carousel={myCarousel}>
    {myCarousel}
    <Pagination /> 
  </InlineGallery>
);
```

IMO the carousel _should_ be the only child that itself has children. And it should also be unique. But I'd much rather these be properties we check on a known carousel rather than properties we use to identify one...

Reply:
This should be resolved with the use of `useContext`